library/src/camera_impl.cxx

```diff
@@ -117,6 +117,60 @@ angle_deg_t camera_impl::getViewAngle()
   return angle;
 }
 
+//----------------------------------------------------------------------------
+angle_deg_t camera_impl::getAzimuth()
+{
+  vtkCamera* cam = this->GetVTKCamera();
+  double pos[3], foc[3];
+  cam->GetPosition(pos);
+  cam->GetFocalPoint(foc);
+  double viewDir[3];
+  vtkMath::Subtract(foc, pos, viewDir);
+  double viewDirProj[3] = { viewDir[0], viewDir[1], 0.0 };
+  double norm = vtkMath::Norm(viewDirProj);
+  if (norm < VTK_DBL_EPSILON)
+  {
+    return 0.0;
+  }
+  vtkMath::Normalize(viewDirProj);
+  return vtkMath::DegreesFromRadians(atan2(viewDirProj[1], viewDirProj[0]));
```

Comment:
Looking at this page for reference https://en.wikipedia.org/wiki/List_of_common_coordinate_transformations#To_spherical_coordinates
While it's important to check that the norm is not zero, I think you can skip the normalization.
You can also check the squared normalization only to save a sqrt for performance reason.
Here's a suggestion (please check it works)

```suggestion
  double viewDirProj[2] = { viewDir[0], viewDir[1] };
  if (vtkMath::SquaredNorm<2>(viewDirProj) < VTK_DBL_EPSILON)
  {
    return 0.0;
  }
  return vtkMath::DegreesFromRadians(atan2(viewDirProj[1], viewDirProj[0]));
```

Reply:
I applied the changes you suggested.